### ISSUE TITLE
"ProfileHelper - Failed to retrieve profile" message in a loop

### ISSUE THREAD

**Issue by cuu508** [AsamK/signal-cli#780]
I have a main signal-cli instance with three linked signal-cli instances (all running 0.9.0). 

Today, I noticed the main signal-cli instance (DBus service) has crashed, but all three linked instances are outputting the following in a loop:

```
Oct 16 15:57:25 www1 signal-cli[3801166]: Got receipt.
Oct 16 15:57:25 www1 signal-cli[3801166]: WARN ProfileHelper - Failed to retrieve profile, ignoring: java.lang.RuntimeException: org.whispersystems.signalservice.api.push.exceptions.RateLimitException: [413] Rate limit exceeded: 413
Oct 16 15:57:25 www1 signal-cli[3801166]: Envelope from: +1[redacted] (device: 2)
Oct 16 15:57:25 www1 signal-cli[3801166]: Timestamp: 1634317366008 (2021-10-15T17:02:46.008Z)
Oct 16 15:57:25 www1 signal-cli[3801166]: Got receipt.
Oct 16 15:57:25 www1 signal-cli[3801166]: WARN ProfileHelper - Failed to retrieve profile, ignoring: java.lang.RuntimeException: org.whispersystems.signalservice.api.push.exceptions.RateLimitException: [413] Rate limit exceeded: 413
Oct 16 15:57:25 www1 signal-cli[3801166]: Envelope from: +1[redacted] (device: 5)
Oct 16 15:57:25 www1 signal-cli[3801166]: Timestamp: 1634317366008 (2021-10-15T17:02:46.008Z)
Oct 16 15:57:25 www1 signal-cli[3801166]: Got receipt.
[...]
```

All the log messages were about one single recipient phone number. And I didn't see any normal incoming messages in the log – signal-cli seemed to be stuck on this one receipt.

I tried stopping the DBus service for a while, trusting the problematic number with "-a", and blocking it. But the same messages kept coming. 

I got things back to normal (I hope) by linking the linked devices again. For each, I

 - removed the data directory
 - ran `signal-cli link`, and `signal-cli addDevice` on the main instance

Now the linked instances seem to be back to normal. 

The main instance is stuck on handling some old messages and eating 100% CPU:

```
Envelope from: unknown source
Timestamp: 1633783955498 (2021-10-09T12:52:35.498Z)
Sent by unidentified/sealed sender
Exception: org.whispersystems.libsignal.InvalidMessageException: invalid message Message decryption failed (ProtocolInvalidMessageException)
No message content

Envelope from: unknown source
Timestamp: 1633783955623 (2021-10-09T12:52:35.623Z)
Sent by unidentified/sealed sender
Exception: org.whispersystems.libsignal.InvalidMessageException: invalid message Message decryption failed (ProtocolInvalidMessageException)
No message content
[...]
```

I'm not sure what's up with that (maybe it has been in a crashed state for a long time and is now working through a backlog?) but will let it continue.

Another weird thing I noticed on the main instance is this out-of-memory crash:

```
Oct 16 16:19:34 www3 signal-cli[491207]: memory allocation of 14663999644663542977 bytes failed
Oct 16 16:19:34 www3 systemd[1]: signal.service: Main process exited, code=killed, status=6/ABRT
Oct 16 16:19:34 www3 systemd[1]: signal.service: Failed with result 'signal'.
```

14663999644663542977 bytes seem an awful lot of memory.

My question – has anybody seen anything similar? Does the above seem like a possible bug, or more likely an operator error? I realize the different error messages may be about unrelated issues, but am posting them here in case they *are* related.

**Comment by cuu508** [comment]
Ah, just noticed 0.9.1 is out with this item in the CHANGELOG:

> Fix issue which could cause signal-cli to repeatedly send the same delivery receipts

Perhaps that's what I ran into?

**Comment by cuu508** [comment]
For the benefit of anybody with a similar problem stumbling on this: I left the main signal-cli instance printing out "Message decryption failed" overnight, but it was still stuck the next day.

I ran `signal-cli receive --verbose` which generated additional output, including session key. 

I ran grep on the signal data directory with the session key, and found the matching phone number in `recipients-store`.

I ran `signal-cli send <phone number> -endsession` a few times (saw that flag in #294).

Now `signal-cli receive` appears to be unstuck and is chewing through the accumulated messages quickly.
